extensions/amp-subscriptions/0.1/entitlement-store.js

```diff
@@ -73,12 +73,29 @@ export class EntitlementStore {
     }
 
     this.firstResolvedPromise_ = new Promise(resolve => {
-      this.onChange(({entitlements}) => {
+      const entitlementsResolved = Object.keys(this.entitlements_).length;
+
+      // Check if current entitlements unblocks the reader
+      for (const key in this.entitlements_) {
+        const entitlements = (this.entitlements_[key]);
         if (entitlements.enablesThis()) {
-          resolve(entitlements);
+          return resolve(entitlements);
         }
-      });
+      }
+
+      if (entitlementsResolved === this.serviceIds_.length) {
+        // Resolve with null if non of the entitlements unblocks the reader
+        return resolve(null);
```

Comment:
I think instead of `null`, we should return the "local" response for now. Eventually, we will need a "decider" callback here, but for now just local should be enough. That also means that we will never resolve `null`.

Reply:
I can only resolve with an `Entitlements` object, so resolving it with a string would just make confusing

Reply:
I didn't mean "local" as a string, sorry. I meant the `Entitlements` object produced by local service. In other words, this promise always yields: it just tries as hard as it can to yield a positive response.

Reply:
Changed this to boolean type